keyboards/silverbullet44/config.h

```diff
@@ -0,0 +1,294 @@
+#define DESCRIPTION     Metalical Keyboard
+
+/* key matrix size */
+#define MATRIX_ROWS 8
+#define MATRIX_COLS 6
+
+/*
+ * Keyboard Matrix Assignments
+ *
+ * Change this to how you wired your keyboard
+ * COLS: AVR pins used for columns, left to right
+ * ROWS: AVR pins used for rows, top to bottom
+ * DIODE_DIRECTION: COL2ROW = COL = Anode (+), ROW = Cathode (-, marked on diode)
+ *                  ROW2COL = ROW = Anode (+), COL = Cathode (-, marked on diode)
+ *
+*/
+#define MATRIX_ROW_PINS { D4, C6, D7, E6 }
+#define MATRIX_COL_PINS { B3, B1, F7, F6, F5, F4 }
+#define UNUSED_PINS
+
+/* COL2ROW, ROW2COL*/
+#define DIODE_DIRECTION COL2ROW
+
+/*
+ * Split Keyboard specific options, make sure you have 'SPLIT_KEYBOARD = yes' in your rules.mk, and define SOFT_SERIAL_PIN.
+ */
+#define SOFT_SERIAL_PIN D2 // or D1, D2, D3, E6
+#define MASTER_RIGHT
+
+// #define BACKLIGHT_PIN B7
+// #define BACKLIGHT_BREATHING
+// #define BACKLIGHT_LEVELS 3
+
+#define RGB_DI_PIN D3
+#ifdef RGB_DI_PIN
+  //#define RGBLIGHT_SPLIT
+  #define RGBLED_SPLIT {26, 26}
+  #define RGBLED_NUM 52
+  #ifdef RGB_MATRIX_ENABLE
+    #define DRIVER_LED_TOTAL 52
+    #define RGB_MATRIX_MAXIMUM_BRIGHTNESS 150
+
+    #define RGB_MATRIX_HUE_STEP 8
+    #define RGB_MATRIX_SAT_STEP 8
+    #define RGB_MATRIX_VAL_STEP 8
+    #define RGB_MATRIX_SPD_STEP 8
+
+    #define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+
+    #define RGB_MATRIX_KEYPRESSES  // reacts to keypresses
+
+    #define RGB_MATRIX_STARTUP_MODE RGB_MATRIX_TYPING_HEATMAP
+  #else
+    #define RGBLIGHT_LED_MAP { 0,1,2,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,3,4,5,6, \
+                              26,27,28,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,29,30,31,32 }
+    #define RGBLIGHT_HUE_STEP 8
+    #define RGBLIGHT_SAT_STEP 8
+    #define RGBLIGHT_VAL_STEP 8
+    #define RGBLIGHT_LIMIT_VAL 150 /* The maximum brightness level */
+  //#define RGBLIGHT_SLEEP  /* If defined, the RGB lighting will be switched off when the host goes to sleep */
+/*== all animations enable ==*/
+//  #define RGBLIGHT_ANIMATIONS
+/*== or choose animations ==*/
+    #define RGBLIGHT_EFFECT_BREATHING
+    #define RGBLIGHT_EFFECT_RAINBOW_MOOD
+    #define RGBLIGHT_EFFECT_RAINBOW_SWIRL
+    #define RGBLIGHT_EFFECT_SNAKE
+//  #define RGBLIGHT_EFFECT_KNIGHT
+//  #define RGBLIGHT_EFFECT_CHRISTMAS
+//  #define RGBLIGHT_EFFECT_STATIC_GRADIENT
+//  #define RGBLIGHT_EFFECT_RGB_TEST
+//  #define RGBLIGHT_EFFECT_ALTERNATING
+/*== customize breathing effect ==*/
+  /*==== (DEFAULT) use fixed table instead of exp() and sin() ====*/
+    #define RGBLIGHT_BREATHE_TABLE_SIZE 256      // 256(default) or 128 or 64
+  /*==== use exp() and sin() ====*/
+    #define RGBLIGHT_EFFECT_BREATHE_CENTER 2     // 1 to 2.7
+    #define RGBLIGHT_EFFECT_BREATHE_MAX    255   // 0 to 255
+/*== customize snake effect ==*/
+    #define RGBLIGHT_EFFECT_SNAKE_LENGTH RGBLED_NUM
+/*== customize knight effect ==*/
+    #define RGBLIGHT_EFFECT_KNIGHT_LENGTH 6
```

Comment:
This block can/should be removed, as it's only for RGBLIGHT, while you have RGB Matrix enabled, using WS2812 for the driver, and these are mutually exclusive. And most of these settings do not carry over. 

This block should be removed, or moved to it's own section.

Reply:
We plan to use RGB MATRIX in the future, but currently only supports RGB LEDs.
I want the user to be able to select in rule.mk.

What does move to own section mean?
Do you mean to surround it with '#ifdef RGBLIGHT_ENABLE'?

Reply:
Above, you hav an `#ifdef RGB_MATRIX_ENABLE`, which means any of the `RGBLIGHT` settings here are ONLY applied if both RGBLIGHT and RGB Matrix is enabled.

Reply:
@drashna 
Isn't SPLIT_KEYBOARD and RGB_MATRIX only work with MASTER now?

I'm waiting for #5998 to be merged.
When # 5998 is merged, enable RGB_MATRIX, so I'd like to keep it as it is until then.